### PR TITLE
(BEDS-469) DA: notifications dashboard table

### PR DESCRIPTION
**Pull request by remoterami**

### CODE REVIEW

File: backend/pkg/api/data_access/notifications.go
Comment (Eisei24) [inline backend/pkg/api/data_access/notifications.go:158]: **suggestion:** You could use `ILIKE` so you don't need to lower the strings.

File: backend/pkg/api/data_access/notifications.go
Comment (Eisei24) [inline backend/pkg/api/data_access/notifications.go:167]: **question:** Why not use this? Is it a `pq.Array` conflict in `NULL` cases or something like that?

**Review by Eisei24 — CHANGES_REQUESTED** [review]
Looks good but please have a look at the cursor issue.

File: frontend/types/api/notifications.ts
Comment (remoterami) [reply frontend/types/api/notifications.ts:42]: Yes, discussed with API: the groups mapping is only present & cached in validator/account dashboards, not the notifications dashboard, so we need to provide group names. The dashboard names mapping is present though so we don't need them, even though it wouldn't have hurt too much

File: backend/pkg/api/data_access/general.go
Comment (remoterami) [reply backend/pkg/api/data_access/general.go:55]: Yeah thought this would help us going forward, glad you like it! I'm also planning to refactor some queries with this.

Feel free to adjust/extend if useful :+1:

File: backend/pkg/api/data_access/notifications.go
Comment (remoterami) [reply backend/pkg/api/data_access/notifications.go:158]: According to my tests a while back `LOWER` seems to be slightly faster, you can also find similar results online (e.g. https://stackoverflow.com/a/20336666). Maybe a preference thing but I'd keep it for that reason

File: backend/pkg/api/data_access/notifications.go
Comment (remoterami) [reply backend/pkg/api/data_access/notifications.go:167]: My initial motivation was that I didn't want to use `pq.StringArray` (because `string` won't work with `ARRAY_AGG`) in the struct type definition because I feared it would mess with the ts struct conversion, but turns out that's not the case - it's way cleaner without `rows.Scan` so I've changed it, thanks for pointing out

File: backend/pkg/api/handlers/common.go
Comment (peterbitfly) [inline backend/pkg/api/handlers/common.go:690]: @LuccaBitfly is this change ok with you?

File: backend/pkg/api/types/notifications.go
Comment (peterbitfly) [inline backend/pkg/api/types/notifications.go:35]: @LuccaBitfly is this change ok with you?

**Review by peterbitfly — COMMENTED** [review]
waiting for green light regarding the changes in the api package, if thats ok the pr can be merged

File: backend/pkg/api/types/data_access.go
Comment (Eisei24) [reply backend/pkg/api/types/data_access.go]: Yes, in the cursor I would have all 4 of them for an easier query.

File: backend/pkg/api/data_access/notifications.go
Comment (Eisei24) [inline backend/pkg/api/data_access/notifications.go:181]: **thought:** Since the default column entries only require those that uniquely identify a row you don't need to add the dashboard and group names to it.
However if you change it this way you would need the `offset` if again.

Both ways work for me but consider that the query could be faster with less WHERE/ORDER BY.

**Review by Eisei24 — APPROVED** [review]
utACK, please wait for the confirmation from other reviewers before merging.
I can't really tell what the linter issue here is.

**Pull request merged**
